google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionImpl.java

```diff
@@ -255,6 +255,7 @@ public ReadContext singleUse(TimestampBound bound) {
             .setRpc(spanner.getRpc())
             .setDefaultQueryOptions(spanner.getDefaultQueryOptions(databaseId))
             .setDefaultPrefetchChunks(spanner.getDefaultPrefetchChunks())
+            .setDefaultDirectedReadOption(spanner.getOptions().getDirectedReadOptions())
```

Comment:
Should this also be set for 'BatchReadOnlyTransactionImpl'?

Reply:
+1
We need to set this option for `BatchReadOnlyTransaction` and also make change in `getExecuteSqlRequestBuilder` and `readInternalWithOptions` of AbstractReadContext class.

Reply:
Added this, as well as additional tests for client level options.